modules/adkernelBidAdapter.js

```diff
@@ -1,310 +1,188 @@
+          data: {
+            zone: Number(zoneId),
+            ad_type: 'rtb',
+            v: VERSION,
+            r: JSON.stringify(request)
           }
         });
-      }, params, AJAX_REQ_PARAMS);
-    }
-
-    /**
-     * Builds complete rtb bid request
-     * @param imps collection of impressions
-     */
-    function buildRtbRequest(imps) {
-      return {
-        'id': utils.getUniqueIdentifierStr(),
-        'imp': imps,
-        'site': site,
-        'at': 1,
-        'device': {
-          'ip': 'caller',
-          'ua': 'caller'
-        }
-      };
-    }
-
-    /**
-     * Build ad-network specific endpoint url
-     */
-    function buildEndpointUrl(host) {
-      return `${window.location.protocol}//${host}/rtbg`;
+      });
+    });
+    return requests;
+  },
+  interpretResponse: function(serverResponse, request) {
+    let response = serverResponse.body;
+    if (!response.seatbid) {
+      return [];
     }
 
-    function buildRequestParams(zone, rtbReq) {
-      return {
-        'zone': encodeURIComponent(zone),
-        'ad_type': 'rtb',
-        'r': encodeURIComponent(JSON.stringify(rtbReq))
+    let rtbRequest = JSON.parse(request.data.r);
+    let rtbImps = rtbRequest.imp;
+    let rtbBids = response.seatbid
+      .map(seatbid => seatbid.bid)
+      .reduce((a, b) => a.concat(b), []);
+
+    return rtbBids.map(rtbBid => {
+      let imp = rtbImps.find(imp => imp.id === rtbBid.impid);
+      let prBid = {
+        requestId: rtbBid.impid,
+        cpm: rtbBid.price,
+        creativeId: rtbBid.crid,
+        currency: 'USD',
+        ttl: 360,
+        netRevenue: true
       };
+      if ('banner' in imp) {
+        prBid.mediaType = BANNER;
+        prBid.width = imp.banner.w;
+        prBid.height = imp.banner.h;
+        prBid.ad = formatAdMarkup(rtbBid);
+      }
+      if ('video' in imp) {
+        prBid.mediaType = VIDEO;
+        prBid.vastUrl = rtbBid.nurl;
+        prBid.width = imp.video.w;
+        prBid.height = imp.video.h;
+      }
+      return prBid;
+    });
+  },
+  getUserSyncs: function(syncOptions, serverResponses) {
+    if (!syncOptions.iframeEnabled) {
+      return [];
     }
-  }
-
-  /**
-   *  Main module export function implementation
-   */
-  baseAdapter.callBids = function (params) {
-    var bids = params.bids || [];
-    processBids(bids);
-  };
-
-  /**
-   *  Process all bids grouped by network/zone
-   */
-  function processBids(bids) {
-    const dispatcher = new RtbRequestDispatcher();
-    // process individual bids
-    utils._each(bids, (bid) => {
-      if (!validateBidParams(bid.params)) {
-        utils.logError(`Incorrect configuration for adkernel bidder: ${bid.params}`);
-        bidmanager.addBidResponse(bid.placementCode, createEmptyBidObject(bid));
-      } else {
-        dispatcher.addImp(bid);
+    let result = [];
+    serverResponses.forEach(response => {
+      if (response.ext && response.ext.adk_usersync) {
```

Comment:
In #1748, the server response format changed here. This was intentional... but I forgot to mention it.

Reply:
@dbemiller done